routers/user/auth.go

```diff
@@ -664,8 +664,23 @@ func LinkAccount(ctx *context.Context) {
 		return
 	}
 
-	ctx.Data["user_name"] = gothUser.(goth.User).NickName
-	ctx.Data["email"] = gothUser.(goth.User).Email
+	uname := gothUser.(goth.User).NickName
+	email := gothUser.(goth.User).Email
+	ctx.Data["user_name"] = uname
+	ctx.Data["email"] = email
+
+	if "" != uname {
```

Comment:
It is better to use `if len(uname) == 0 {`

Reply:
I haven’t seen that pattern before in other Go code. What’s the rationale?

Reply:
@lafriks bump

Reply:
(I know you've changed this already I just figure you'd like to have a reason for this rather than a "handwavy" comment)